src/libs/actions/Session/index.ts

```diff
@@ -331,8 +331,17 @@ function signInWithShortLivedAuthToken(email: string, authToken: string) {
                 isLoading: true,
             },
         },
+        // We are making a temporary modification to 'signedInWithShortLivedAuthToken' to ensure that 'App.openApp' will be called at least once
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.SESSION,
+            value: {
+                signedInWithShortLivedAuthToken: true,
+            },
+        },
     ];
 
+    // Subsequently, we revert it back to the default value of signedInWithShortLivedAuthToken (undefined) in 'successData' to ensure the user is logged out on refresh
     const successData: OnyxUpdate[] = [
```

Comment:
Currently, `successData` and `failureData` is the same object. Could we store it in a single variable, like `completedData`?

I don't think we have a conventional name for such a thing, "completed" is the best thing that came to my mind. Could also be `resultData`?

Reply:
We can do something like this: 

```
const commonData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: ONYXKEYS.ACCOUNT,
            value: {
                isLoading: false,
            },
        },
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: ONYXKEYS.SESSION,
            value: {
                signedInWithShortLivedAuthToken: undefined,
            },
        },
    ];
    
    const successData = [...commonData];
    const failureData = [...commonData];
```
    
        
What do you think?

Reply:
We can do something like this: 

```
const commonData: OnyxUpdate[] = [
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: ONYXKEYS.ACCOUNT,
            value: {
                isLoading: false,
            },
        },
        {
            onyxMethod: Onyx.METHOD.MERGE,
            key: ONYXKEYS.SESSION,
            value: {
                signedInWithShortLivedAuthToken: undefined,
            },
        },
    ];
    
    const successData = [...commonData];
    const failureData = [...commonData];
```
    
        
What do you think?

Reply:
Sure, we can use completed instead of common since whether or not it was success or failure it's still completed so it would make sense

Reply:
I guess my only wonder here is, would we never add something different for successData within signInWithShortLivedAuthToken?

Reply:
`doneData` ? `finalData` ?

Reply:
It's literal copy-paste of a dozen of lines. I think it's obvious that they should be merged. We could even pass `failureData: successData` with a comment like `// Failure data and success data are same in this case`.

I'm fine with `resolutionData`, I'm also fine with `failureData: successData`, but please de-duplicate this. If success data and failure data start to differ in the future, it can by trivially changed.

Reply:
@cubuspl42 Are you suggesting to pass `resolutionData` alongside `optimisticData` without defining `successData` and `failureData` seperately? I'm asking because I'm not very familure with how expensify is handling success and failure scenarious after the API call is made

Reply:
Let me know what you think of the current changes

Reply:
Looks good 👍